idseq_dag/util/s3.py

```diff
@@ -146,58 +163,33 @@ def fetch_from_s3(src,  # pylint: disable=dangerous-default-value
     untar = auto_untar and dst.lower().endswith(".tar")
     if untar:
         dst = dst[:-4]  # Remove .tar
-    abspath = os.path.abspath(dst)
 
+    destdir = os.path.dirname(dst)
+    tmp_destdir = os.path.join(destdir, "tmp_downloads")
+    tmp_dst = os.path.join(tmp_destdir, os.path.basename(dst))
+
+    abspath = os.path.abspath(dst)
```

Comment:
Concurrent attempts to fetch the same destination will serialize on the destination lock (a different lock for each destination file --- that way we can have multiple destinations being downloaded concurrently, but only one download at a time for a given destination).    When the first of the concurrent downloads for destination succeeds, the rest would have waited patiently for that, and would then return immediately. 

Once a file is downloaded, it cannot and will not be modified by this session.